chainstate/tx-verifier/src/transaction_verifier/timelock_check.rs

```diff
@@ -139,5 +138,40 @@ pub fn check_timelocks<
         }
     }
 
+    let outputs = match tx.outputs() {
+        Some(outputs) => outputs,
+        None => return Ok(()),
```

Comment:
This is really dangerous and wrong. Here's why:

This function isn't only for this match and relevant checks and doesn't strictly depend on it. The problem here is with the following scenario: Imagine someone in the future adding more functionality after the for loop below for something other than the outputs. What's gonna happen is basically that his functionality will be limited by whether the outputs are Some or None. This is certainly not the desired behavior by default.

The correct way to do this should not limit the contiuation of the function UNLESS the whole function is about this check. Meaning: You can either create another function that will do the check AND run the loop and return if unnecessary (like you did in this match), or you limit what you did here to a smaller control block that doesn't affect the whole function in case someone adds more things to be done in the future.